modules/xcm-bridge-hub-router/src/lib.rs

```diff
@@ -0,0 +1,395 @@
+// Copyright 2019-2021 Parity Technologies (UK) Ltd.
+// This file is part of Parity Bridges Common.
+
+// Parity Bridges Common is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Parity Bridges Common is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Parity Bridges Common.  If not, see <http://www.gnu.org/licenses/>.
+
+//! Pallet that may be used instead of `SovereignPaidRemoteExporter` in the XCM router
+//! configuration. The main thing that the pallet offers is the dynamic message fee,
+//! that is computed based on the bridge queues state. It starts exponentially increasing
+//! if the queue between this chain and the sibling/child bridge hub is congested.
+//!
+//! All other bridge hub queues offer some backpressure mechanisms. So if at least one
+//! of all queues is congested, it will eventually lead to the growth of the queue at
+//! this chain.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use bp_xcm_bridge_hub_router::LocalXcmChannel;
+use codec::Encode;
+use frame_support::traits::Get;
+use sp_runtime::{traits::One, FixedPointNumber, FixedU128, Saturating};
+use xcm::prelude::*;
+use xcm_builder::{ExporterFor, SovereignPaidRemoteExporter};
+
+pub use pallet::*;
+pub use weights::WeightInfo;
+
+pub mod benchmarking;
+pub mod weights;
+
+mod mock;
+
+/// The factor that is used to increase current message fee factor when bridge experiencing
+/// some lags.
+const EXPONENTIAL_FEE_BASE: FixedU128 = FixedU128::from_rational(105, 100); // 1.05
+/// The factor that is used to increase current message fee factor for every sent kilobyte.
+const MESSAGE_SIZE_FEE_BASE: FixedU128 = FixedU128::from_rational(1, 1000); // 0.001
+
+/// Maximal size of the XCM message that may be sent over bridge.
+///
+/// This should be less than the maximal size, allowed by the messages pallet, because
+/// the message itself is wrapped in other structs and is double encoded.
+pub const HARD_MESSAGE_SIZE_LIMIT: u32 = 32 * 1024;
+
+/// The target that will be used when publishing logs related to this pallet.
+pub const LOG_TARGET: &str = "runtime::bridge-hub-router";
```

Comment:
True, but all other (bridge) pallet log targets are starting with `runtime::bridge`, so you can get all related logs using `RUST_LOG=runtime::bridge=trace`. OTOH xcm logs are using `xcm::` prefix like `xcm::currency_adapter`, `xcm::barriers` and so on. So I wonder if maybe we should make it `xcm::bridge-hub-router` or something like that? Or `runtime::bridge` is fine? WDYT?

Reply:
True, I see. Then probably something that starts with `bridge` would be better. But then the fact that the log target differs from the pallet name would be confusing to me. If the pallet name is `pallet-xcm-bridge-hub-router`, I would expect the log target to be `xcm-bridge-hub-router`. I honestly don't know what would be better, but I don't have a strong preference here and it's not a blocker.

Reply:
On a second though, maybe `xcm::` prefix is better for this pallet - it will be deployed at asset hubs (and other sending chains - other bridge pallets are deployed at bridge hubs.